src/games/strategy/triplea/ai/proAI/util/ProTransportUtils.java

```diff
@@ -162,6 +162,25 @@ public int findUnitsTransportCost(final List<Unit> units) {
     return transportCost;
   }
 
+  public List<Unit> getAirThatCantLandOnCarrier(final PlayerID player, final Territory t, final List<Unit> units) {
+    final GameData data = ai.getGameData();
+    int capacity = AirMovementValidator.carrierCapacity(units, t);
+    final Collection<Unit> airUnits = Match.getMatches(units, ProMatches.unitIsAlliedAir(player, data));
+    List<Unit> airThatCantLand = new ArrayList<Unit>();
+    for (final Unit airUnit : airUnits) {
```

Comment:
Should we sort the air unit list first to try and prefer to fit as many of the smaller aircraft as possible? I'm thinking for example Cold War where helo and fighter have the same carrier cost but different defense values. There may be other maps too with "fat" planes that take up the whole carrier not leaving room for the small ones that maybe can defend better. 

Just a consideration, the context may not care if the units that can't land are the best defenders or not.